elastic/apm-server
# Update network connection fields

## Motivation/summary

The OTel semantic conventions for network connection type have evolved: `net.host.connection_type` is now `net.host.connection.type` and `net.host.connection.subtype`. See https://github.com/open-telemetry/opentelemetry-specification/pull/1759. This PR extends https://github.com/elastic/apm-server/pull/5436 to adapt to these changes.

Support Elastic APM agents sending "network.connection.type" ("n.c.t" in RUMv3) in metadata, which will be added to all events in the stream. This is currently intended for RUM. In the future we may want to support setting this at the even level too, to cater for operation-specific network information.

Finally, the Elasticsearch field names have been changed to `network.connection.type` and `network.connection.subtype`.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Run the iOS agent
2. Check that the `network.connection.type` and `network.connection.subtype` fields are recorded

## Related issues

https://github.com/elastic/apm-server/issues/5203
